orchestra/workflow/load.py

```diff
@@ -193,11 +197,19 @@ def load_workflow_version(version_data, workflow, force=False):
 def _verify_dependencies_not_updated(step_data, dependency_attr,
                                      old_dependencies):
     new_dependencies = set(step_data.get(dependency_attr, []))
-    if old_dependencies is not None and old_dependencies != new_dependencies:
+    old_set = set(old_dependencies or [])
+    new_set = set(new_dependencies)
+    if old_dependencies is not None and (new_set - old_set):
```

Comment:
`old_dependencies is not None` feels unnecessary because of your logic on l.200?